docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,365 @@
+      * [Deletion](#deletion)
+    * [Use cases](#use-cases)
+      * [Recovery from permanent quorum loss](#recovery-from-permanent-quorum-loss)
+      * [Trigger on-demand snapshot compaction](#trigger-on-demand-snapshot-compaction)
+      * [Trigger on-demand full/delta snapshots](#trigger-on-demand-fulldelta-snapshots)
+      * [Trigger on-demand maintenance of etcd cluster](#trigger-on-demand-maintenance-of-etcd-cluster)
+      * [Copy backups from one object store to another object store](#copy-the-backups-from-one-object-store-to-another-object-store)
+  * [Metrics](#metrics)
+
+## Summary
+
+This DEP proposes an enhancement to `etcd-druid`'s capabilities to handle [out-of-band](#terminology) tasks, which are presently performed manually or invoked programmatically via sub-optimal APIs. The document proposes the establishment of an unified interface by defining a well-structured API to harmonize the initiation of any `out-of-band` task, monitor its status, and simplify the process of adding new tasks and managing their lifecycle.
+
+## Terminology
+
+* **druid:** [etcd-druid](https://github.com/gardener/etcd-druid) is an operator to manage the etcd clusters.
+
+* **backup-sidecar:** It is the etcd-backup-restore sidecar container running in each etcd-member pod of etcd cluster.
+
+* **leading-backup-sidecar:** A backup-sidecar that is associated to an etcd leader of an etcd cluster.
+
+* **out-of-band task** Any on-demand tasks/operations that can be executed on an etcd cluster without modifying the Etcd custom resource spec (desired state).
+
+## Motivation
+
+Today, [Etcd-druid](https://github.com/gardener/etcd-druid) mainly acts as an etcd cluster provisioner(creation, maintenance and deletion). In future, capabilities of etcd-druid will be enhanced via [etcd-member](https://github.com/gardener/etcd-druid/blob/8ac70d512969c2e12e666d923d7d35fdab1e0f8e/docs/proposals/04-etcd-member-custom-resource.md) proposal by providing it access to a much more detailed information about each etcd cluster member. While we enhance the reconciliation and monitoring capabilities of etcd-druid, it still lacks an ability to provide an easy-to-use handle for an operator to trigger `out-of-band` tasks on an existing etcd cluster.
+
+There are new learnings while operating etcd clusters at scale. It has been observed that we regularly need capabilities to trigger `out-of-band` tasks which are out of the purview of a regular reconciliation run. Replacing manual intervention by an operator to invoke any `out-of-band` tasks, thus alleviating any mistakes.
+Some examples of an `on-demand/out-of-band` operations:
+
+* Recover from a permament quorum loss.
+* Trigger an on-demand full/delta snapshot.
+* Trigger an on-demand snapshot compaction.
+* Trigger an on-demand maintenance of etcd cluster.
+* Copy the backups from one object store to another object store.
+* Inspect full/delta snapshots to gauge DB size growth and identify potential hot-spots.
```

Comment:
we can remove this. I thought about it a bit more and this does not fit into the task framework but it surely can be provided as a `druidctl` sub-command. The reason is that the result of such an activity could result in a LOT of data and we should not store this as part of the task status.

Reply:
make sense